src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -1,16 +1,61 @@
 package com.synopsys.integration.detect.workflow.componentlocationanalysis;
 
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
 import com.synopsys.integration.blackduck.api.generated.view.DeveloperScansScanView;
-import java.util.List;
+import com.synopsys.integration.fixpr.generic.beans.Component;
+
+import java.util.*;
 
 /**
  * Transforms a list of {@link DeveloperScansScanView} to a list of {@link Component}s, which will then be used to
  * assemble the input to Component Locator.
  *
- * Will be fully implemented in a subsequent pull request to the Fix PR feature branch.
  */
 public class ScanResultToComponentListTransformer {
-    public ComponentLocatorInput transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
-        return new ComponentLocatorInput(null, null, null);
+    /**
+     * Given a list of reported components from a Rapid/Stateless Detector scan, transforms each element to its
+     * corresponding {@link Component} with appropriate metadata.
+     * @param rapidScanFullResults
+     * @return list of {@link Component}s
+     */
+    public List<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
+        HashMap<String, ScanMetadata> componentIdWithMetadata = new HashMap<>(); // TODO investigate if duplicates may make sense here?
```

Comment:
Duplicates should be pruned.

Reply:
Ok, removed this TODO comment but did add the question to this tech debt jira ticket IDETECT-3950